badger_api/requests.py

```diff
@@ -64,3 +65,13 @@ def fetch_all_claimable_balances(chain: str):
             data = future.result()["rewards"]
             results = {**results, **data}
     return results
+
+
+@lru_cache
+def fetch_token_names(chain: str):
+    return requests.get(f"{badger_api}/tokens?chain={chain}").json()
```

Comment:
Add in the retry library https://findwork.dev/blog/advanced-usage-python-requests-timeouts-retries-hooks/ and handle retries here gracefully